packages/google_maps_flutter/google_maps_flutter_android/android/src/main/java/io/flutter/plugins/googlemaps/CozyMarkerBuilder.java

```diff
@@ -82,56 +42,141 @@ private float getTextXOffset(float markerWidth, Rect rect) {
+
+        /* setting colors */
+        int defaultMarkerColor = Color.WHITE;
+        int defaultTextColor = Color.BLACK;
+        int defaultIconCircleColor = Color.rgb(248, 249, 245);
+
+        int selectedMarkerColor = Color.rgb(57, 87, 189);
+        int selectedTextColor = Color.WHITE;
+        int selectedIconCircleColor = Color.WHITE;
+
+        int visualizedMarkerColor = Color.rgb(248, 249, 245);
+        int visualizedTextColor = Color.rgb(110, 110, 100);
+        int visualizedIconCircleColor = Color.WHITE;
+
+        int markerColor = defaultMarkerColor;
+        int textColor = defaultTextColor;
+        int iconCircleColor = defaultIconCircleColor;
+
+        if(markerData.isVisualized) {
+            markerColor = visualizedMarkerColor;
+            textColor = visualizedTextColor;
+            iconCircleColor = visualizedIconCircleColor;
+        }
+        if(markerData.isSelected) {
+            markerColor = selectedMarkerColor;
+            textColor = selectedTextColor;
+            iconCircleColor = selectedIconCircleColor;
+        }
+
+        /* setting constants */
+        // setting global constants
+        int paddingVertical = Math.round(getDpFromPx(12));
+        int paddingHorizontal = Math.round(getDpFromPx(11.5f));
+        int minMarkerWidth = Math.round(getDpFromPx(40));
+        int strokeSize = Math.round(getDpFromPx(1.5f));
+
+        // setting constants for pointer
+        int pointerWidth = Math.round(getDpFromPx(7));
+        int pointerHeight = Math.round(getDpFromPx(6));
+        
+        // setting constants for icon
+        int iconSize =  Math.round(getDpFromPx(16));
+        int iconCircleSize =  Math.round(getDpFromPx(24));
+        int iconLeftPadding =  Math.round(getDpFromPx(5));
+        int iconRightPadding =  Math.round(getDpFromPx(3));
+
+        /* setting variables */
         // gets the text size based on the font
-        Rect rect = new Rect();
+        Rect textBounds = new Rect();
         float textSize = getDpFromPx(12f);
         Paint priceMarkerTextStyle = getTextPaint(textSize, textColor);
-        priceMarkerTextStyle.getTextBounds(text, 0, text.length(), rect);
+        priceMarkerTextStyle.getTextBounds(text, 0, text.length(), textBounds);
 
-        // set the marker width
-        int paddingVertical = Math.round(getDpFromPx(12f));
-        int paddingHorizontal = Math.round(getDpFromPx(11f));
-        int minMarkerWidth = Math.round(getDpFromPx(40f));
-        int strokeSize = Math.round(getDpFromPx(1.5f));
-        int markerWidth = rect.width() + (2 * paddingHorizontal) + strokeSize;
+        // getting icon bitmap
+        Bitmap iconBitmap = getIconBitmap(markerData.icon,iconSize,iconSize);
+
+        // additionalIconWidth to be used on markerWidth
+        int iconAdditionalWidth = iconBitmap != null ? iconCircleSize + iconRightPadding : 0;
+
+        // pointerSize to be used on bitmap creation
+        int pointerSize = (hasPointer ? pointerHeight : 0);
```

Comment:
This is related to the little triangle we add to the marker, right? I think pointer might be too generic to understand here, what do you think of using some more specific naming? I searched a little bit and on wikipedia they name this a "tail", which to me at least sounds a little bit more specific. We could switch from `pointerSize` to `markerTailHeight`. What do you think?

Reply:
Good point! It was tail, I just changed to be consistent with cozy contract. I will check with duda actually if it makes more sense to change the contract to `hasTail` instead of `hasPointer`.

Reply:
Talking to her, she said it is more clear to design team what pointer means. I will leave it as pointer to not have double naming and prevent more confusion